dialog/fileitem.go

```diff
@@ -76,52 +77,53 @@ func (f *fileDialog) newFileItem(location fyne.URI, dir, up bool) *fileDialogIte
 }
 
 type fileItemRenderer struct {
-	item *fileDialogItem
+	item         *fileDialogItem
+	fileTextSize float32
 
 	icon    *widget.FileIcon
 	text    *widget.Label
 	objects []fyne.CanvasObject
 }
 
-func (s fileItemRenderer) Layout(size fyne.Size) {
+func (s *fileItemRenderer) Layout(size fyne.Size) {
 	if s.item.picker.view == gridView {
 		s.icon.Resize(fyne.NewSize(fileIconSize, fileIconSize))
 		s.icon.Move(fyne.NewPos((size.Width-fileIconSize)/2, 0))
 
 		s.text.Alignment = fyne.TextAlignCenter
-		s.text.Resize(fyne.NewSize(size.Width, fileTextSize))
-		s.text.Move(fyne.NewPos(0, size.Height-s.text.MinSize().Height))
+		s.text.Resize(fyne.NewSize(size.Width, s.fileTextSize))
+		s.text.Move(fyne.NewPos(0, size.Height-s.fileTextSize))
 	} else {
 		s.icon.Resize(fyne.NewSize(fileInlineIconSize, fileInlineIconSize))
 		s.icon.Move(fyne.NewPos(theme.Padding(), (size.Height-fileInlineIconSize)/2))
 
 		s.text.Alignment = fyne.TextAlignLeading
-		s.text.Resize(fyne.NewSize(size.Width, fileTextSize))
-		s.text.Move(fyne.NewPos(fileInlineIconSize, (size.Height-s.text.MinSize().Height)/2))
+		textMin := s.text.MinSize()
+		s.text.Resize(fyne.NewSize(size.Width, textMin.Height))
+		s.text.Move(fyne.NewPos(fileInlineIconSize, (size.Height-textMin.Height)/2))
 	}
 	s.text.Refresh()
 }
 
-func (s fileItemRenderer) MinSize() fyne.Size {
-	var padding fyne.Size
-
+func (s *fileItemRenderer) MinSize() fyne.Size {
 	if s.item.picker.view == gridView {
-		padding = fyne.NewSize(fileIconCellWidth-fileIconSize, theme.Padding())
-		return fyne.NewSize(fileIconSize, fileIconSize+fileTextSize).Add(padding)
+		return fyne.NewSize(fileIconCellWidth, fileIconSize+s.fileTextSize)
 	}
 
-	padding = fyne.NewSize(theme.Padding(), theme.Padding()*4)
-	return fyne.NewSize(fileInlineIconSize+s.text.MinSize().Width, fileTextSize).Add(padding)
+	textMin := s.text.MinSize()
+	return fyne.NewSize(fileInlineIconSize+textMin.Width+theme.Padding(), textMin.Height)
 }
 
-func (s fileItemRenderer) Refresh() {
+func (s *fileItemRenderer) Refresh() {
+	s.fileTextSize = widget.NewLabel("M\nM").MinSize().Height // cache two-line label height
```

Comment:
Just wandering if this can be improved in any way? Seems a bit strange to create a new temporary label just to get the minsize but I understand if there isn't a better way.

Reply:
I agree that it doesn't look neat - but I tried the alternative of keeping a widget around and checking the MinSize regularly and it was slower.